tests/p2p/test_protocol.py

```diff
@@ -251,21 +293,24 @@ def test_invalid_same_peer_id2(self) -> None:
         # one of the peers will close the connection. We don't know which one, as it depends
         # on the peer ids
 
-        if self.conn.tr1.disconnecting or self.conn.tr2.disconnecting:
-            conn_dead = self.conn
+        if bytes(self.peer1.id) > bytes(self.peer2.id):
+            tr_dead = self.conn.tr1
+            tr_dead_value = self.conn.peek_tr1_value()
+            proto_alive = conn.proto2
             conn_alive = conn
-        elif conn.tr1.disconnecting or conn.tr2.disconnecting:
-            conn_dead = conn
-            conn_alive = self.conn
         else:
-            raise Exception('It should never happen.')
-        self._check_result_only_cmd(conn_dead.peek_tr1_value() + conn_dead.peek_tr2_value(), b'ERROR')
+            tr_dead = conn.tr2
+            tr_dead_value = conn.peek_tr2_value()
+            proto_alive = self.conn.proto1
+            conn_alive = self.conn
+
+        self._check_result_only_cmd(tr_dead_value, b'ERROR')
         # at this point, the connection must be closing as the error was detected on READY state
-        self.assertIn(True, [conn_dead.tr1.disconnecting, conn_dead.tr2.disconnecting])
-        # check connected_peers
-        connected_peers = list(self.manager1.connections.connected_peers.values())
-        self.assertEquals(1, len(connected_peers))
-        self.assertIn(connected_peers[0], [conn_alive.proto1, conn_alive.proto2])
+        self.assertTrue(tr_dead.disconnecting)
+        # check ready_peers
+        ready_peers = list(self.manager1.connections.iter_ready_connections())
+        self.assertEquals(1, len(ready_peers))
+        self.assertEquals(ready_peers[0], proto_alive)
```

Comment:
Since I refactored the method that chooses which connection to drop, I updated this test to actually test the specific connection that is dropped, which is important.

Before, it would just test that _some_ connection is dropped, but not _which exact_ connection is dropped.